iothub_client/inc/iothub_device_client.h

```diff
@@ -355,7 +355,10 @@ extern "C"
     *           This function is expected to be used along with:
     *           `IoTHubDeviceClient_CreateUploadContext`
     *           `IoTHubDeviceClient_AzureStoragePutBlock`
+    *           `IoTHubDeviceClient_AzureStoragePutBlockList`
```

Comment:
I would rather move the list of how-to things either in the DevDoc or in a header (a single block, instead of copy/pasting everything) then referencing that doxygen block.